### PR TITLE
Use RabbiMQ pub sub library to queue the deposit tx from eth - cennznet

### PR DESCRIPTION
**Pull request by KarishmaBothara**
Deposit claim on eth  - [Queue tx ] -> subscriber submits the tx on CENNZnet after # event confirmation->   [ Queue the submission] -> subscriber on verification updates the db with successful/invalid entry

### CODE REVIEW

File: scripts/subscribeEthereumDeposit.js [hunk 1]
```diff
@@ -7,8 +7,9 @@ const mongoose = require('mongoose');
 const { BridgeClaim  } = require('../src/mongo/models');
 const ethers = require('ethers');
 const { curly } = require("node-libcurl");
-const { hexToU8a } = require("@cennznet/util");
+const { hexToU8a } = require("@polkadot/util");
 const pegAbi = require("../abi/ERC20Peg.json").abi;
+const PubSub = require('pubsub-js');
```
Comment (holygits) [inline scripts/subscribeEthereumDeposit.js:12]: how does pubsub js manage persistence? what happens to the queues if the service restarts?

Comment (KarishmaBothara) [reply scripts/subscribeEthereumDeposit.js:12]: that's a good point.. we already are managing persistence via mongo db.. on restart, I will fill these queues with the data from db, (eth confirm/cennznet confirm) status in db, add to the queue

**Review by holygits — COMMENTED** [review]
direction looks good. needs tests

File: scripts/subscribeEthereumDeposit.js [hunk 2]
```diff
@@ -76,6 +100,135 @@ async function sendClaim(claim, transactionHash, api, signer, nonce) {
...
+            console.log('bridgeClaimRecord:::',bridgeClaimRecord);
+            const eventDetails = await ClaimEvents.find({_id: bridgeClaimRecord.txHash});
+            console.log('eventDetails::',eventDetails);
+            if (eventDetails.length > 0) {
+                const claim = {
+                    tokenAddress: eventDetails[0].tokenAddress,
+                    amount: eventDetails[0].amount,
+                    beneficiary: eventDetails[0].beneficiary
+                };
+                console.log('claim:::::::::',claim);
```
[the remark this answers is not in the document]
Comment (KarishmaBothara) [reply scripts/subscribeEthereumDeposit.js:200]: makes it easy to debug... but I will remove it...

**Comment by eupston** [comment]
- switches deposit subscription to RabbitMQ Pub/Sub service
- Adds Ethereum poller to catch any missed deposit events
- create retry mechanism for messages if they fail
- alerts slack channel when all retries fail
- add tests for subscription and poller service
- updates testing env in CI

File: scripts/subscribeEthereumDeposit.js [hunk 3]
```diff
@@ -97,66 +204,200 @@ async function main (networkName, pegContractAddress) {
...
-        await updateTxStatusInDB('EthereumConfirming', eventInfo.transactionHash, null, cennznetAddress);
-        const tx = await eventInfo.getTransaction();
-        await tx.wait(eventConfirmation + 1);
+        await handleDepositEvent(api, eventInfo.transactionHash, cennznetAddress, amount, tokenAddress, eventConfirmations, channel );
+    });
+}
+
+async function handleDepositEvent(api, transactionHash, cennznetAddress, amount, tokenAddress, eventConfirmations, channel ) {
+    const checkIfBridgePause = await api.query.ethBridge.bridgePaused();
+    if (!checkIfBridgePause.toHuman()) {
```
Comment (KarishmaBothara) [inline scripts/subscribeEthereumDeposit.js:232]: we don't need to worry about bridge pause while depositing...
https://github.com/cennznet/bridge-contracts/pull/70
Relayer should still be able to deposit a claim..

Comment (eupston) [reply scripts/subscribeEthereumDeposit.js:232]: gotcha will update that

File: scripts/ethereumEventPoller.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            logger.error("Mongo DB disconnected Poller stopping...");
+            break;
+        }
+        const allEvents = await peg.queryFilter({});
+        const depositEventTxHashes = allEvents.filter(event => event.event === "Deposit").map(event => event.transactionHash);
+        //check if any deposit tx hashes are not in DB yet
+        const allDepositClaims = await BridgeClaim.find({});
+        const allDepositClaimsTxHashes = allDepositClaims.map(claim => claim.txHash);
+        //get any events in db that were submitted when bridge paused
+        const pausedClaims = allDepositClaims.filter(claim => claim.status === "Bridge Paused");
```
Comment (KarishmaBothara) [inline scripts/ethereumEventPoller.js:56]: there won't be "Bridge Paused" status as we still claim the deposit

**Review by aidan-starke — APPROVED** [review]

**Pull request merged**